### PR TITLE
Add keep to drop duplicates

### PR DESCRIPTION
**Pull request by phofl**
ref #525

### CODE REVIEW

**Review by rjzamora — COMMENTED** [review]
Just a quick note that I'm skeptical that `keep="first"` and `keep="last"` are actually "working".  When we perform the shuffle, there is no explicit guarantee that the original ordering will be preserved. I have a suspicion that we are just getting lucky.

**Comment by phofl** [comment]
We generally preserve row order with task based or p2p shuffle, but wouldn’t guarantee this

**Comment by rjzamora** [comment]
>We generally preserve row order with task based or p2p shuffle, but wouldn’t guarantee this

Right exactly. I do think "tasks" will always preserve order, but it seems like "p2p" may depend on the order that messages are received/buffered by a worker (maybe/hopefully I'm wrong about this?).

**Comment by rjzamora** [comment]
**Update**: Ran some small tests, and it does look like "p2p" will produce wrong results here:

```python
import dask
from dask.distributed import LocalCluster, Client
import dask.dataframe as dd
from dask.datasets import timeseries

client = Client(LocalCluster())
shuffle = "p2p"

df = timeseries()
result_pd = df.compute().drop_duplicates(subset=["name"], keep="first")
result_dd = df.drop_duplicates(subset=["name"], keep="first", split_out=df.npartitions, shuffle=shuffle).compute()

# Fails for shuffle = "p2p"
# Passes for shuffle = "tasks"
dd.assert_eq(result_pd, result_dd)
```

I realize this is more of a `dask.dataframe` issue than a `dask-expr` issue. I'll raise an issue over there to discuss how this behavior should be handled and/or communicated.

**Comment by hendrikmakait** [comment]
Just to assert what you already figured out empirically: P2P shuffling has no ordering guarantees.

**Comment by rjzamora** [comment]
>I am not too concerned about this, users can still request split_out=1

Yeah, I'm assuming we have similar feelings, but just to be explicit. The two things we definitely want to avoid:

1. We silently return incorrect results
2. The code crashes

My loosely-held opinion on this: In the short term, we should adopt `how=None` and warn the user if they use `how="first"/"last"` when "p2p" is active. In the long term, figure out a simple/reliable way to **optionally** enforce stable ordering in "p2p" for cases like this.

File: dask_expr/_collection.py [hunk 1]
```diff
@@ -1622,13 +1639,26 @@ def nunique(self, dropna=True):
         else:
             return uniqs.size
 
-    def drop_duplicates(self, ignore_index=False, split_every=None, split_out=True):
+    def drop_duplicates(
+        self, ignore_index=False, split_every=None, split_out=True, keep="first"
```
Comment (milesgranger) [inline dask_expr/_collection.py:1643]: Curious why keep=None in DataFrame and keep="first" in Series?

**Review by milesgranger — APPROVED** [review]
One question, but otherwise looks goodie!

File: dask_expr/_collection.py (hunk 1 above)
Comment (phofl) [reply dask_expr/_collection.py]: that was a mistake, thanks for catching

**Comment by milesgranger** [comment]
Thanks @phofl!

**Pull request merged**